### ISSUE TITLE
Wrong file layout when unsing exports and exports sources

### ISSUE THREAD

**Issue by claasd** [conan-io/conan#981]
Hi,

In some special cases, conan creates a wrong file layout in the source and export folder.
The problem occurs when exports exports a file from a subdirectory and export_sources exports files from this directory as well. Then, the directory is created again inside the existing directory. The error dos NOT occur when using the source command, only when using install or test_package.

I added a test project 
[export_sources_bug.zip](https://github.com/conan-io/conan/files/773716/export_sources_bug.zip)

The test_package simply imports the file layout from the package. 
Result:
```
src
| tools
| src
     | tools
     | data
```

Expected:
```
src
| tools
| data
```

Environment: Windows, conan 0.19.1 installed via pip, Python 3.5.2

**Comment by memsharded** [comment]
I am having a look at it. It seems the problem is that by definition ``exports`` and ``exports_sources`` must not overlap. In this case, your patterns they are both exporting the same ``include.py`` file, twice, and this is causing some confusion. 

I am double-checking, because even in that case, I don't fully understand what is happening with the folders

**Comment by claasd** [comment]
Hi,
you can change the example to a way that exports and exports_sources do not overlap, same result.
```python 
exports_sources = "src/data/*"
```

**Comment by memsharded** [comment]
Yes, there is a bug in the way that nested folders are managed. We are opening a minor release, and I am working on a fix, we might try to include it, though it needs some new tests. Thanks for your report!

**Comment by claasd** [comment]
It is a rare case, and we have a workaround. 
For me, it can be included in a later release.

**Comment by memsharded** [comment]
Released in 0.19.2, please check. Thanks for reporting!
